File: Dataset/Leetcode/valid/104/57.cpp
class Solution {
public:
    int XXX(TreeNode* root) {
        int l = 0;
        int r = 0;
        if (root == nullptr) return 0;
        else {
            l = XXX(root->left) + 1;
            r = XXX(root->right) + 1;
        }
        return l > r ? l: r;
    }
};

File: Dataset/Leetcode/valid/100/57.py
 class Solution:
    def XXX(self, p: TreeNode, q: TreeNode) -> bool:
        #先判断是否全为空，是则返回；再判断是否有一个为空，是则返回，再判断两个值是否相等
        if not p and not q:
            return True
        elif p == None or q == None:
            return False
        elif p.val != q.val:
            return False
        return self.XXX(p.left,q.left) and self.XXX(p.right,q.right)

